src/download-artifact.ts

```diff
@@ -80,23 +83,36 @@ async function run(): Promise<void> {
       latest: true,
       ...options
     })
+    artifacts = listArtifactResponse.artifacts
 
-    if (listArtifactResponse.artifacts.length === 0) {
-      throw new Error(
-        `No artifacts found for run '${inputs.runID}' in '${inputs.repository}'`
+    core.debug(`Found ${artifacts.length} artifacts in run`)
+
+    if (inputs.pattern) {
```

Comment:
The else block of `if (isSingleArtifactDownload)` above,
https://github.com/actions/download-artifact/blob/f44cd7b40bfd40b6aa1cc1b9b5b7bf03d3c67110/src/download-artifact.ts#L77-L80
should probably be moved as an else block of the `if` on this line.

Currently it prints a slightly confusing message when specifying a `pattern` input: https://github.com/nulano/Pillow/actions/runs/7281978318/job/19844364694#step:3:36